### DISCUSSION TITLE
Booking.com / Expedia Integration

### DISCUSSION THREAD

**Discussion by seagullua** [organicmaps/organicmaps#626] [Ideas]
I really like the feature, when you can see the prices of the hotel as long as have a quick link to the booking site.

It is a cool feature when you travel and need to find a place to stay. Also, it should be able to generate a significant revenue stream to support the project.

**Comment by GLLM** [comment]
Well, I understand this may look appealing in some context, but this is a downvote for me.

Too much tracking on the internetz nowadays ... let's keep FOSS free of this.

Well, that's clearly my own personal opinion !
Thanks
GLLM

**Reply by seagullua** [reply]
Third party services integration will not break privacy at all. The backend can be used which will make calls to third party service, so expedia / booking com will know nothing about the user.

Also, the dialog can popup, which asks if you want to use this feature.

I am not asking about booking system inside the app. I am just asking about placing all available hotels onto the map, like we gas stations etc.

**Comment by biodranik** [comment]
It requires a proprietary deal with booking com and Internet access. Why not using the Booking app instead?

**Reply by seagullua** [reply]
Expedia.com has api and will easily open it when requested.

**Reply by seagullua** [reply]
I am not asking about booking system inside the app. I am just asking about placing all available hotels onto the map, like we have gas stations etc.

**Comment by j5k** [comment]
If you really want to use tracking in your app -> use MAPS.ME. This app should stay privacy focused.

**Reply by seagullua** [reply]
Third party services integration will not break privacy at all. The backend can be used which will make calls to third party service, so expedia / booking com will know nothing about the user.

Also, the dialog can popup, which asks if you want to use this feature.

Answer to your question. I was fan of Maps.ME for years. However, it is become more broken with each update. Recently, I have lost all my pins, which I collected for years, just because of their bug in online sync. After the latest acquisition the things seem to become even worse. So, I am looking for an alternative.

**Reply by biodranik** [reply]
The idea is to motivate people to add hotels and other objects directly into the OpenStreetMap.

**Comment by mpawelski** [comment]
Here's my two cents.

I see here quite a big resentment to the idea of having booking.com data in Organic Maps.
But I think this is mostly because the audience here is very FOSS and privacy focused. Most people don't care though. They just want to use functional "maps app".
And the fact that Maps.me have booking.com accommodations is invaluable. I never booked anything from Maps.met app. I used it mostly in these scenarios:

1. You have already booked hotel/hostel through booking.com website. You arrived to the city and want to know where to go to the hotel.
2. You are in the new place and want to walk around to check different hostels in the area.

Scenario 1. is the most valuable for me. Something I use all the time when I travel. And Maps.me is huge among travelers (almost everyone I met used it when I traveled Southeast Asia) and I bet this booking.com integration is very valuable to them too.

And we don't need to compromise on privacy. Like @seagullua said:
>I am not asking about booking system inside the app. I am just asking about placing all available hotels onto the map, like we have gas stations etc.

Plus possibility to get this hotels in search result. That's all that we need. Booking system inside the app is not that useful anyway.

Actually, I think the way Maps.met shows booking.com objects is too "seamless". For example we could have some subtle indication next to hotel icon that it's from booking.com (like small "B" sing next to it). This will make it clear that this is not OSM data.

> The idea is to motivate people to add hotels and other objects directly into the OpenStreetMap.

This is very worthy idea, but the fact is that there are many places around the world where you have hotel/hostels from booking.com and there are not added to OSM. And the end user might not care that much why it is like that, he might even not know what is OpenStreetMap. And those people might still be very passionate about the Organic Maps project and even donate to it.

As a final note, I want to be able to clearly recommend Organic Maps to other users of Maps.me and say that "this is better that Maps.me in every way. It updates map much more regularly and displays it better. bla bla bla..." but the lack of booking.com data there will always force me to add "just keep in mind that it might not have hotels from Booking.com" because I know how valuable it is to someone who travels a lot.

**Reply by biodranik** [reply]
You need to make a private deal with Booking.com to get their data. We can discuss the integration if you help with that "minor" issue )
I highly doubt that Booking.com share its data for free without integration of the booking system.

**Comment by Esmolante** [comment]
I'm a newcomer here (so far I've been using the old Igo 8 running on Windows CE, still a fantastic program today but no Android version). I've tried a dozen other navigators on Android over a few weeks and why I've stuck with this one is not because it adheres to FOSS (I like it), but to KISS. I prefer a program that performs a single function, but does it well, to an all-purpose program loaded with heavy modules.
There's a saying in Spanish: "He who embraces much, presses little".
Let's make an excellent program that will guide us to our destination. Deciding what our destination is going to be is already out of its competence. 
Show the gas stations or hotels in an area? That's perfect. Show their prices? It adds unnecessary complication. There are other tools for that. IMHO.

**Comment by mnalis** [comment]
- 2 years later, this (quite disliked here) idea is being implemented in https://github.com/organicmaps/organicmaps/pull/6523  (Kayak == booking.com)
